server/models/pages.js

```diff
@@ -926,6 +927,29 @@ module.exports = class Page extends Model {
     return rebuildJob.finished
   }
 
+  /**
+   *
+   * @returns
+   */
+  static async buildXMLSitemap() {
+    const xmlTree = [
+      '<?xml version="1.0" encoding="UTF-8"?>',
+      '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">'
+    ]
+
+    const pages = await WIKI.models.pages.query().select(['path', 'updatedAt']).where('isPublished', '=', true)
```

Comment:
You load all pages regardless of their permissions. Pages should match what the Guest group can see.

Reply:
Gotcha!